compiler/lib/js_output.ml

```diff
@@ -1979,19 +1980,20 @@ let program ?(accept_unnamed_var = false) f ?source_map p =
     | Some sm ->
         let sources = hashtbl_to_list files in
         let names = hashtbl_to_list names in
-        let sources_content =
-          match contents with
-          | None -> None
-          | Some r -> Some (List.rev !r)
+        let sources_contents =
+          let open Option.Syntax in
+          let* r = contents in
```

Comment:
I would avoid introducing `let*` syntax here to respect the style of codebase.

Just use the following maybe
```ocaml
Option.map (fun x -> List.rev !x) contents
```